src/Services/PortfolioPlanningDataService.ts

```diff
@@ -76,8 +80,159 @@ export class PortfolioPlanningDataService {
         return client.runGetQuery(fullQueryUrl).then(
             (results: any) => this.ParseODataWorkItemQueryResultResponse(results),
             (error) => this.ParseODataErrorResponse(error));
+    }
+
+    public async GetAllPortfolioPlans() : Promise<PortfolioPlanningDirectory>
```

Comment:
I need to refactor this class to separate extension storage services from OData-related services. I'll do that later in a separate PR - not really important now.